### PR TITLE
Add rest endpoints for Book

### LINKED ISSUE #463: Add a new REST endpoint to manage Book resource
**This feature is to enable the UI migration from Vaadin to JS**
A new REST endpoint should be able to:

- Add a new book
- Update an existing book
- Get an existing book
- Delete an existing book

**Describe the solution you'd like**
The solution would be to introduce a new REST endpoint for managing Book resource. The endpoint would expose the APIs to achieve below:

- Save a new book into the db
- Update an existing book
- Search a book by some criteria
- Delete and existing book

The media type for request and response should be in JSON format. There shouldn't be any breaking changes in service and dao layer so that there is no impact on the existing Vaadin UI.

### PR DESCRIPTION
**Pull request by kateerfle**
## Summary of change

Added BookController.java

## Related issue

Closes #463 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-jcijyenp-JiKFGBv62FIPoFnvOW6Ubg)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,106 @@
 {
   "dependencies": {
-    "@material-ui/core": "^4.11.2"
+    "@material-ui/core": "^4.11.2",
+    "lit-element": "^2.2.1",
+    "@polymer/polymer": "3.2.0",
+    "@webcomponents/webcomponentsjs": "^2.2.10",
+    "@vaadin/vaadin-grid": "5.7.7",
```
Comment (knjk04) [inline package.json:7]: We shouldn't have any Vaadin dependencies in `0.2.0`. Did you get this from `main`?

Comment (kateerfle) [reply package.json:7]: I branched off of 0.2.0, so I really don't know why I have those Vaadin dependencies in my branch.

Comment (kateerfle) [reply package.json:7]: I just checked my local 0.2.0 branch to make sure the dependencies didn't contain Vaadin, and that branch was fine (Vaadin-free).  I branched off of that local branch to create my feature branch, so I have no idea how this happened (I probably merged incorrectly when I tried to push to my remote branch).  I am trying to push the correct package.json file up now

Comment (kateerfle) [reply package.json]: Ok, the package.json file is fixed now.

**Pull request merged**